core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -473,7 +479,14 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     }
 
     def commit() {
-      consumer.commitSync(offsets.map { case (tp, offset) =>  (tp, new OffsetAndMetadata(offset, ""))}.asJava)
+      val existingTopics: Set[String] = Try(consumer.listTopics) match {
+        case Success(allTopics) => allTopics.keySet.asInstanceOf[Set[String]]
+        case Failure(_) => Set.empty
+      }
+      if (existingTopics.nonEmpty)
+        consumer.commitSync(offsets.filterKeys(tp => existingTopics.contains(tp.topic)).map { case (tp, offset) => (tp, new OffsetAndMetadata(offset, "")) }.asJava)
```

Comment:
nit: we can use ```OffsetAndMetadata(long offset)``` constructor

Reply:
Also, split to multiple lines. Typical line width is 120 characters.